### PR TITLE
Write results from perfscale to google sheet

### PR DESCRIPTION
**Pull request by paigerube14**
Want to write results to a sheet to be able to more easily analyze results 

Need to still find the proper google sheet credentials that has API accesss

### CODE REVIEW

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
Need to update sandman tool to find workload type from output (jira ticket for tracking: https://issues.redhat.com/browse/OCPQE-15494)

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-router-perf-gcp

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

File: ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  from_image:
+    namespace: ci
+    name: ocp-qe-perfscale-ci
+    tag: latest
+  commands: openshift-qe-write-perfscale-results-commands.sh
+  credentials:
+  - namespace: test-credentials # this entry injects the custom credential
+    name: ocp-qe-perfscale-es
+    mount_path: /secret
+  - namespace: test-credentials # this entry injects the custom credential
```
Comment (paigerube14) [inline ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-ref.yaml:12]: @qiliRedHat please let me know what the correct namespace and name are here. Doesn't look to be correct

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

File: ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-ref.yaml (hunk 1 above)
Comment (qiliRedHat) [reply ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-ref.yaml:12]: @paigerube14 I added the credential for sa-google-sheet here https://vault.ci.openshift.org/ui/vault/secrets/kv/show/selfservice/openshift-qe/sa-google-sheet
  "secretsync/target-name": "sa-google-sheet",
  "secretsync/target-namespace": "test-credentials,ci"

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-ci-tests

**Comment by paigerube14** [comment]
Properly printing results to spreadsheet here: https://docs.google.com/spreadsheets/d/1cciTazgmvoD0YBdMuIQBRxyJnblVuczgbBL5xC8uGuI/edit#gid=84336777
Need to make updates in write-results-to-sheet to not only look for flexy clusters 
https://github.com/openshift-qe/ocp-qe-perfscale-ci/blob/e0e9a960b76ab526e43b200e3c256160387451c4/write_to_sheet/write_helper.py#L56

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.14-ocp-qe-perfscale-node-density-heavy-aws

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.14-ocp-qe-perfscale-node-density-heavy-aws periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-router-perf-gcp periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-gcp-4.13-ocp-qe-perfscale-ci-tests-net-perf

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.14-ocp-qe-perfscale-node-density-heavy-aws periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-router-perf-gcp periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-gcp-4.13-ocp-qe-perfscale-ci-tests-net-perf

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.14-ocp-qe-perfscale-node-density-heavy-aws periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-ocp-qe-perfscale-router-perf-gcp periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-gcp-4.13-ocp-qe-perfscale-ci-tests-net-perf

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-gcp-4.13-ocp-qe-perfscale-ci-tests-net-perf

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-gcp-4.13-ocp-qe-perfscale-ci-tests-net-perf

**Comment by paigerube14** [comment]
@svetsa-rh @qiliRedHat can you ptal at this pr. I added the ability to print results to a prow specific google sheet using a chain on each workload type

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-gcp-4.13-ocp-qe-perfscale-ci-tests-net-perf

**Review by svetsa-rh — CHANGES_REQUESTED** [review]
@paigerube14 Not sure if I am missing something. 
1) I see some scripts with only one line invoking script from parent dir. And some other workloads refer to the script from parent dir directly.
2) I see new step code for writing results for some workloads written separately, but most of the code looks same. And some workloads don't have this.
It might be easier for maintenance if all workloads are pointing to same.

**Comment by paigerube14** [comment]
We are calling the openshift-qe-write-perfscale-results-commands in all cases. Because cluster-density-v2 and node density are running in the same workflow in some configurations, it didnt allow me to use the same ref multiple times in one workflow. So I created sub folders for the workloads that could be run together in a workflow but each commands script just references back to the same write script. Hopefully that clarifies things

File: ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-commands.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+sa_email=$(jq -r .client_email ${GSHEET_KEY_LOCATION})
+echo "$sa_email"
+
+export EMAIL_ID_FOR_RESULTS_SHEET=prubenda@redhat.com
+
+mkdir sandman
+
+git clone https://github.com/openshift-qe/ocp-qe-perfscale-ci.git sandman -b main
+git clone https://github.com/paigerube14/ocp-qe-perfscale-ci.git -b prow_write
```
[the remark this answers is not in the document]
Comment (paigerube14) [reply ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-commands.sh:33]: pr to get this merged: https://github.com/openshift-qe/ocp-qe-perfscale-ci/pull/483

**Comment by svetsa-rh** [comment]
> Because cluster-density-v2 and node density are running in the same workflow in some configurations

Can you clarify which workflows?

It seems to me like the current approach is a loop and redundant. May be we should look at which workflows or configurations are not allowing us to go with simpler approach?

File: ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-commands.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (svetsa-rh) [reply ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-commands.sh:33]: Can we migrate this prow_write to Prow and make it a reusable step that others can leverage too?

**Comment by paigerube14** [comment]
@svetsa-rh https://github.com/openshift/release/blob/91a4c5a2b854a5a0e5532312f021a0184fcedf36/ci-operator/step-registry/openshift-qe/ocp-qe-perfscale-ci-tests/openshift-qe-ocp-qe-perfscale-ci-tests-workflow.yaml
The main workflow for kube-burner tests

File: ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-commands.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (paigerube14) [reply ci-operator/step-registry/openshift-qe/write-perfscale-results/openshift-qe-write-perfscale-results-commands.sh:33]: right now all of the code is specific to our tests and prints to our defined google sheet that has write access from the google sheet sa account. Think that could be a future effort. I was trying to reuse as much of the code that's already written for our jenkins implementation. If we move this one code into prow it wont have the proper imports until you clone the rest of the ocp-qe-perfscale-ci repo

**Comment by svetsa-rh** [comment]
> @svetsa-rh https://github.com/openshift/release/blob/91a4c5a2b854a5a0e5532312f021a0184fcedf36/ci-operator/step-registry/openshift-qe/ocp-qe-perfscale-ci-tests/openshift-qe-ocp-qe-perfscale-ci-tests-workflow.yaml The main workflow for kube-burner tests

I really want to move away from this main workflow. lol. Seems to add to more redundancy and convoluted code in other places. Have we verified how must gather is being collected when 2 tests are running in same chain and also 2nd test won't run if 1st test fails right?

**Comment by paigerube14** [comment]
@svetsa-rh I think for tests we know consistently run and run well I dont think we need to move away from the workflow, I think it will give us more results with less cluster costs if we get it consistently passing (which it has been pretty consistent)
The must gather will be run at the very end of a test no matter pass or one test fails

**Comment by svetsa-rh** [comment]
> @svetsa-rh I think for tests we know consistently run and run well I dont think we need to move away from the workflow, I think it will give us more results with less cluster costs if we get it consistently passing (which it has been pretty consistent) The must gather will be run at the very end of a test no matter pass or one test fails

I assume must-gather will always be based off of 2nd test, irrespective of either tests pass or fail.

**Comment by paigerube14** [comment]
@svetsa-rh If first test fails, the second test doesnt run in a chain so the must gather will have that data. i think it collects all events during the life of the cluster so we should be able to see both test information if pass(not 100% on that)

**Comment by svetsa-rh** [comment]
@paigerube14 If we change configuration/workflows for Self managed clusters in future, cleaning up and pointing to same step for all workloads will be good.

**Pull request merged**